mongodm.go

```diff
@@ -266,8 +264,7 @@ func (self *Connection) Register(document IDocumentBase, collectionName string)
 
 	//check if model was already registered
 	if _, ok := self.modelRegistry[typeName]; !ok {
-
-		collection := self.database.C(collectionName)
+		collection := self.Session.DB("").C(collectionName) // empty string returns db name from dial info
```

Comment:
Basically, if an `""` is passed into the `DB` function, it will return the database name that was given during the dial info. Here is a quote from the mgo docs: 


> DB returns a value representing the named database. If name is empty, the database name provided in the dialed URL is used instead. If that is also empty, "test" is used as a fallback in a way equivalent to the mongo shell.